hydra-main/src/main/java/com/addthis/hydra/job/spawn/balancer/SpawnBalancer.java

```diff
@@ -1185,14 +1213,46 @@ protected void updateAggregateStatistics(Iterable<HostState> hosts) {
             updateActiveJobIDs();
             double maxMeanActive = -1;
             double maxDiskPercentUsed = -1;
+            long maxDiskFree = 0;
+            long minDiskFree = Long.MAX_VALUE;
+            long sumDiskFree = 0;
+            double maxTaskPercent = 0;
+            double minTaskPercent = 1;
+            double sumTaskPercent = 0;
             for (HostState host : hosts) {
                 maxMeanActive = Math.max(maxMeanActive, host.getMeanActiveTasks());
                 maxDiskPercentUsed = Math.max(maxDiskPercentUsed, getUsedDiskPercent(host));
             }
             for (HostState host : hosts) {
-                cachedHostScores.put(host.getHostUuid(),
-                                     calculateHostScore(host, maxMeanActive, maxDiskPercentUsed));
+                // metrics here
+                HostScore score = calculateHostScore(host, maxMeanActive, maxDiskPercentUsed);
+                cachedHostScores.put(host.getHostUuid(), score);
+                long freeDisk = score.getFreeDiskBytes();
+                double taskPercent = score.getScoreValue(false);
+                sumDiskFree += freeDisk;
+                sumTaskPercent += taskPercent;
+                maxDiskFree = Math.max(freeDisk, maxDiskFree);
+                minDiskFree = Math.min(freeDisk, minDiskFree);
+                maxTaskPercent = Math.max(taskPercent, maxTaskPercent);
+                minTaskPercent = Math.min(taskPercent, minTaskPercent);
             }
+            // update average metrics
+            int numScores = cachedHostScores.size();
+            long avgDiskFree = sumDiskFree / (long) numScores;
+            double avgTaskPercent = sumTaskPercent / (double) numScores;
+            long sumDiskFreeDiff = 0;
+            double sumTaskPercentDiff = 0;
+            for (HostScore score : cachedHostScores.values()) {
+                long diskDiff = Math.abs(avgDiskFree - score.getFreeDiskBytes());
+                sumDiskFreeDiff += diskDiff;
+                sumTaskPercentDiff += Math.abs(avgTaskPercent - score.getScoreValue(false));
+            }
+            avgDiskFreeDiff = sumDiskFreeDiff / (long) numScores;
+            avgTaskPercentDiff = sumTaskPercentDiff / (double) numScores;
+            minDiskFreeDiff = avgDiskFree - minDiskFree;
+            maxDiskFreeDiff = maxDiskFree - avgDiskFree;
```

Comment:
No changes have been made to the core logic of HostScore. It uses heuristics developed by Al, but they are based on the average percentage use of task slots and average percentage use of disk space.

There are also more than a few configuration parameters that can be adjusted, and something like excess disk space has myriad solutions. Many of which are better than potentially overloading a minion's other resources.